### ISSUE TITLE
Unable to save multiple screenshots of the same site — time is not included in the file name

### ISSUE THREAD

**Issue by gentoo-root** [mozilla-services/screenshots#4229]
When I make a screenshot of a site and save it locally, it names the file using the following pattern:

Screenshot-YYYY-M-D site title.png

No time is included in this pattern, and there is no configurable option to change the pattern. When I make multiple screenshots of the same page or of different pages with the same title, they all are saved under the same name, and I need to change it manually to avoid overwriting the files. If the pattern contained the current time (with seconds) in addition to the current date, no filename conflicts would happen.

It also would be nice to have the possibility to customize the naming pattern (e.g. I'd prefer to remove "Screenshot" word at all and to have a different date format, i.e. 20170315 instead of 2017-3-15).

**Comment by chenba** [comment]
This is a duplicate of #4211.

@gentoo-root  What filesystem are you using?

**Comment by gentoo-root** [comment]
>What filesystem are you using?

btrfs, but it's filesystem-independent.

>This is a duplicate of #4211.

Not exactly. #4211 doesn't mention the feature request to add customizable patterns.

Still, (1), (2), etc. mentioned in #4211 look uglier than timestamps which would make naming more consistent (i.e. all filenames have date and time in contrast to having a digit appended starting with the second file).

**Comment by chenba** [comment]
We haven't been able to reproduce the issue with files being overwritten (on macOS, Windows 10, or Linux).  An integer in parentheses is automatically appended to the filename. E.g. 'Screenshot_2018-03-15 Firefox Screenshots.png', 'Screenshot_2018-03-15 Firefox Screenshots(1).png', 'Screenshot_2018-03-15 Firefox Screenshots(2).png', and so on.

What OS distro and version, Firefox release channel, and Firefox version are you using?

**Comment by gentoo-root** [comment]
I've tested it again, and it really does append "(1)", "(2)" etc. to the filename, just as you said. Sorry for the invalid bug description and not having tested it once again before reporting. I might have remembered the behavior incorrectly, or it might have been the case for some older version, sorry for making noise about that.

Consider my report to be a feature request, because in my opinion, it's much nicer to have these names:

```
Screenshot-2018-3-16-12-34-56 mozilla-services screenshots.png
Screenshot-2018-3-16-12-35-23 mozilla-services screenshots.png
Screenshot-2018-3-16-12-35-45 mozilla-services screenshots.png
```

instead of these:

```
Screenshot-2018-3-16 mozilla-services screenshots.png
Screenshot-2018-3-16 mozilla-services screenshots(1).png
Screenshot-2018-3-16 mozilla-services screenshots(2).png
```

"(1)" etc. looks to the user as if the file with the same name had been copied to the directory from different directories, and the user had decided to keep both versions so that "(1)" had been appended. Those screenshot files aren't perceived as a sequence, because the first one has no number in the end in contrast to the others.

And it would be even better if I could customize the pattern and do something like this:

```
20180316 12:34:56 mozilla-services screenshots.png
20180316 12:35:23 mozilla-services screenshots.png
20180316 12:35:45 mozilla-services screenshots.png
```

>What OS distro and version, Firefox release channel, and Firefox version are you using?

I'm on Gentoo Linux, using Firefox stable 58.0.1 (59 isn't in the portage tree yet, I'll update as soon as it appears there).

**Comment by wresuolc** [comment]
The appended numbers are the operating system standard these days, so I'm going to wontfix this.  Thanks for filing and following up though.

**Comment by gentoo-root** [comment]
OK. I still have a few questions, and I would be grateful if some of you answered them.

Should I file another issue for my feature request? As far as I see, it has been ignored here although I've mentioned it three times. It's not directly connected to the naming issue; customizable name patterns would be a nice feature to have irrespective of it.

What's the point of having date at all in the screenshot name? Operating systems of these days store modification dates and times for all the files. Why should the date be included into the file name? The only thing I could think of is sorting, but it's totally broken if the time isn't there.

For example, if I took some screenshots of different sites, and both date and time were in the filename, the screenshots would be sorted in the chronological order:
```
20180319 12:39:04 Site C.png
20180319 12:39:15 Site B.png
20180320 12:34:56 Site A.png
```
And if we don't include time, we get just a random order:
```
20180319 Site B.png
20180319 Site C.png
20180320 Site A.png
```
So why put the date into the filename at all?

**Comment by wresuolc** [comment]
Sorry I didn't call it out specifically, but we talked about it in our triage meeting.  The custom naming patterns is a good idea, it's just a matter of priorities.  We have a couple of full time people working on this project, 282 open issues, and a backlog of features we'd like to implement.  If we look at number of people who will use, for example, Firefox Accounts integration to sync their screenshots between devices vs the number of people who would change their naming pattern, it seems like we should put our efforts towards the FxA integration.  In a perfect world, we'd do everything, but as it stands, we've got to focus on a couple of enhancements and skip the rest.
